src/validator/GroupByValidator.cpp

```diff
@@ -116,10 +123,131 @@ Status GroupByValidator::validateGroup(const GroupClause *groupClause) {
+            if (!groupVisitor.found()) {
+                return Status::SemanticError("Yield non-agg expression `%s` must be"
+                " functionally dependent on items in GROUP BY clause", expr->toString().c_str());
+            }
+        }
+
+        if (!yieldCols_.empty()) {
+            std::unordered_set<Expression*> yieldSet(begin(yieldCols_), end(yieldCols_));
+            FindAnySubExprVisitor yieldVisitor(yieldSet, false);
+            for (auto* expr : groupKeys_) {
+                expr->accept(&yieldVisitor);
+                if (!yieldVisitor.found()) {
+                    return Status::SemanticError("GroupBy item `%s` must be"
+                    " in Yield list", expr->toString().c_str());
+                }
+            }
+        } else {
+            return Status::SemanticError("GroupBy list must in Yield list");
+        }
+    }
+    return Status::OK();
+}
+
+Status GroupByValidator::rewriteInnerAggExpr(YieldColumn* col, bool& rewrited) {
+    auto colOldName = deduceColName(col);
+    auto* colExpr = col->expr();
+    // agg col need not rewrite
+    DCHECK(colExpr->kind() != Expression::Kind::kAggregate);
+    auto collectAggCol = colExpr->clone();
+    auto aggs = ExpressionUtils::collectAll(collectAggCol.get(),
+                                            {Expression::Kind::kAggregate});
+    if (aggs.size() > 1) {
+        return Status::SemanticError("Agg function nesting is not allowed: `%s`",
+                                     collectAggCol->toString().c_str());
+    }
+    if (aggs.size() == 1) {
+        auto colRewrited = colExpr->clone();
+        // set aggExpr
+        col->setExpr(aggs[0]->clone().release());
+        auto aggColName = col->expr()->toString();
+        // rewrite to VariablePropertyExpression
+        RewriteAggExprVisitor rewriteAggVisitor(new std::string(),
+                                                new std::string(aggColName));
+        colRewrited->accept(&rewriteAggVisitor);
+        rewrited = true;
+        needGenProject_ = true;
+        projCols_->addColumn(new YieldColumn(colRewrited.release(),
+                             new std::string(colOldName)));
+    }
+
+    return Status::OK();
+}
+
+Status GroupByValidator::checkAggExpr(AggregateExpression* aggExpr) {
+    auto func = aggExpr->name();
+    if (!func) {
+        return Status::SemanticError("`%s` aggregate function not set.",
+                                     aggExpr->toString().c_str());
+    }
+
+    auto iter = AggregateExpression::NAME_ID_MAP.find(func->c_str());
+    if (iter == AggregateExpression::NAME_ID_MAP.end()) {
+        return Status::SemanticError("Unkown aggregate function `%s`", func->c_str());
+    }
+
+    auto* aggArg = aggExpr->arg();
+    if (graph::ExpressionUtils::findAny(aggArg,
+                                        {Expression::Kind::kAggregate})) {
+        return Status::SemanticError("Agg function nesting is not allowed: `%s`",
```

Comment:
modify the `Agg` to the whole word when as error msg.

Reply:
Got it.